docs/src/manuals/sharpbits/rule-not-found.md

```diff
@@ -0,0 +1,139 @@
+
+- Ensures fast and reliable inference when rules exist
+- Avoids potential numerical instabilities from approximations
+- Throws an error when analytical solutions don't exist
+
+This means you may encounter `RuleNotFoundError` even in cases where approximate solutions could theoretically work. This is intentional - RxInfer will tell you explicitly when you need to consider alternative approaches (like those described in the Solutions section below) rather than silently falling back to potentially slower or less reliable approximations.
+
+## Visualizing the message passing graph
+
+To better understand where message passing rules are needed, let's look at a simple factor graph visualization:
+
+```mermaid
+graph LR
+    %% Other parts of the graph
+    g1[g] -.-> x
+    h1[h] -.-> z
+    y -.-> g2[p]
+    
+    %% Main focus area
+    x((x)) -.- m1[["μ<sub>x→f</sub>"]] --> f[f]
+    f --> m2[["μ<sub>f→y</sub>"]] -.- y((y))
+    z((z)) -.- m3[["μ<sub>z→f</sub>"]] --> f
+
+    %% Styling
+    classDef variable fill:#b3e0ff,stroke:#333,stroke-width:2px;
+    classDef factor fill:#ff9999,stroke:#333,stroke-width:2px,shape:square;
+    classDef otherFactor fill:#ff9999,stroke:#333,stroke-width:2px,opacity:0.3;
+    classDef message fill:none,stroke:none;
+    class x,y,z variable;
+    class f factor;
+    class g1,g2,h1 otherFactor;
+    class m1,m2,m3 message;
+```
+
+In this example:
+- Variables (`x`, `y`, `z`) are represented as circles
+- The factor node (`f`) is represented as a square
+- Messages (μ) flow along the edges between variables and factors, with subscripts indicating direction (e.g., x→f flows from x to f)
+- Faded nodes (g, h) represent other parts of the factor graph that aren't relevant for this local message computation
+
+To compute the outgoing message `f→y`, RxInfer needs:
+1. Rules for how to process incoming messages `x→f` and `z→f`
+2. Rules for combining these messages based on the factor `f`'s type
+3. Rules for producing the outgoing message type that `y` expects
+
+A `RuleNotFoundError` occurs when any of these rules are missing. For example, if `x` sends a `Normal` message but `f` doesn't know how to process `Normal` inputs, or if `f` can't produce the type of message that `y` expects.
+
+## Solutions
+
+### 1. Convert to conjugate pairs
+
+First, try to reformulate your model using conjugate prior-likelihood pairs. Conjugate pairs have analytical solutions for message passing and are well-supported in RxInfer. For example, instead of using a Normal likelihood with Beta prior on its precision, use a Normal-Gamma conjugate pair. See [Conjugate prior - Wikipedia](https://en.wikipedia.org/wiki/Conjugate_prior#Table_of_conjugate_distributions) for a comprehensive list of conjugate distributions.
```

Comment:
You've been using code blocks for distributions like `Normal`, `Beta` everywhere but here 😄 .